From f8eee37712926fdde46c669a7e0b6e438b04134b Mon Sep 17 00:00:00 2001
From: Zhou weixin <zwx@rock-chips.com>
Date: Thu, 18 Jan 2018 16:31:36 +0800
Subject: [PATCH] ARM: dts: rockchip: detect sdcard by gpio for rk312x

if the sdmmc detect with iomux sdmmc_detn will false trigger
once the system into deepsleep

Change-Id: I5928b94327175ee2b39bd0740ea2eafa9d28604d
Signed-off-by: Weixin Zhou <zwx@rock-chips.com>
---
 arch/arm/boot/dts/rk3126-bnd-d708.dts | 2 +-
 arch/arm/boot/dts/rk312x.dtsi         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/rk3126-bnd-d708.dts b/arch/arm/boot/dts/rk3126-bnd-d708.dts
index 2a9e53405897..3440b00af3e4 100644
--- a/arch/arm/boot/dts/rk3126-bnd-d708.dts
+++ b/arch/arm/boot/dts/rk3126-bnd-d708.dts
@@ -582,10 +582,10 @@
 &sdmmc {
 	cap-mmc-highspeed;
 	supports-sd;
-	broken-cd;
 	card-detect-delay = <800>;
 	ignore-pm-notify;
 	keep-power-in-suspend;
+	cd-gpios = <&gpio2 7 GPIO_ACTIVE_LOW>; /* CD GPIO */
 	status = "disabled";
 };
 
diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index ff15186cbef8..d8c376684f66 100644
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -1247,7 +1247,7 @@
 			};
 
 			sdmmc_cmd: sdmmc-cmd {
-				rockchip,pins = <1 RK_PC1 1 &pcfg_pull_default>;
+				rockchip,pins = <1 RK_PB7 1 &pcfg_pull_default>;
 			};
 
 			sdmmc_wp: sdmmc-wp {
-- 
2.35.3

